pandas/core/indexes/multi.py

```diff
@@ -389,6 +390,9 @@ def set_levels(self, levels, level=None, inplace=False,
                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
                    names=[u'foo', u'bar'])
         """
+        if is_list_like(levels) and not is_subscriptable(levels):
+            levels = list(levels)
+
         if level is not None and not is_list_like(level):
             if not is_list_like(levels):
```

Comment:
you can just add
```
elif is_list_like(levels):
   levels = list(levels)
```

Reply:
Tried this approach, but unfortunately, it breaks functionality where one passes to set_levels CategoricalIndex (or in general, something else which is more complex than a list).

Specifically, this turns

```python
index.set_levels(CategoricalIndex(list("bac")), 0)
```

into 

```python
index.set_levels(list("bac"), 0)
```

and breaks [test_set_levels_categorical](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexes/multi/test_get_set.py#L385)

I can add change this with 

```python
if is_list_like(levels) and not isinstance(levels, Index):
```

but I'm not sure what's better

Reply:
the 2nd would be fine, we don't need/want any more api checkers)